doc/design/dist_train/README.md

```diff
@@ -0,0 +1,117 @@
+# Distributed Training Design Doc
+
+## Goal
+
+### Easy To Use
```

Comment:
I think we need a section describing the architecture:

1. the introduction of the master process,
1. the master dispatches tasks to trainer processes,
1. trainer processes exchange gradients via parameter servers.

Also, as we talked before, to track the tasks, the master process need to maintain three queues:

1. todo
1. pending
1. done

Initially, all tasks are in the todo queue. Once a task is dispatched to a trainer, it's moved into the pending queue. Once the trainer reports that it completes the task, it's then moved to the done queue. Or, if a dispatched task times out, i.e., the trainer didn't report completion in expected time, the task is moved from the pending queue back to the todo queue for another dispatching.